packages/at_secondary_server/test/sync_unit_test.dart

```diff
@@ -0,0 +1,724 @@
+        await syncProgressiveVerbHandler.processVerb(
+            response, syncVerbParams, atConnection);
+        List syncResponseList = jsonDecode(response.data!);
+        expect(syncResponseList[0]['atKey'], 'public:phone.wavi@alice');
+        expect(syncResponseList[0]['operation'], '-');
+      });
+      tearDown(() async => await tearDownMethod());
+    });
+
+    group('A group of tests on TTL and TTB with respect to sync', () {
+      setUp(() async => await setUpMethod());
+      test(
+          'test to verify when TTL of a key is expired and deleted then commit operation should have delete',
+          () async {
+        /// Preconditions:
+        /// 1. The key is created on server at time T1 with TTL set
+        /// 2. The key is expired at time T2 (T2 > T1); and key is deleted from keystore
+        /// 3. The sync triggers at time T3 (T3 > T2)
+        /// 4. The key is deleted and commitLog should have a commitOp.delete
+        ///
+        /// Assertions:
+        /// 1. The sync response should contain the commit entry of commitOp.delete
+        await secondaryPersistenceStore!.getSecondaryKeyStore()?.put(
+            'public:lastname.wavi@alice',
+            AtData()
+              ..data = '8897896765'
+              ..metaData = (AtMetaData()..ttl = 1));
+        // manually trigger the deleteExpiredKeys to remove the expired keys
+        await secondaryPersistenceStore!
+            .getSecondaryKeyStore()
+            ?.deleteExpiredKeys();
+        var syncProgressiveVerbHandler = SyncProgressiveVerbHandler(
+            secondaryPersistenceStore!.getSecondaryKeyStore()!);
+        var response = Response();
+        var inBoundSessionId = '_6665436c-29ff-481b-8dc6-129e89199718';
+        var atConnection = InboundConnectionImpl(null, inBoundSessionId);
+        atConnection.metaData.isAuthenticated = true;
+        var syncVerbParams = HashMap<String, String>();
+        syncVerbParams.putIfAbsent(AT_FROM_COMMIT_SEQUENCE, () => '-1');
+        await syncProgressiveVerbHandler.processVerb(
+            response, syncVerbParams, atConnection);
+        List syncResponseList = jsonDecode(response.data!);
+        expect(syncResponseList[0]['atKey'], 'public:lastname.wavi@alice');
+        expect(syncResponseList[0]['operation'], '-');
+      });
+
+      test(
+          'Arch - test to verify when TTL of a key is expired but not deleted when sync triggers',
+          () {
+        /// Preconditions:
+        /// 1. The key is created on server at time T1
+        /// 2. The key is expired at time T2 (T2 > T1); but key is still in keystore and not deleted
+        /// 3. The sync triggers at time T3 (T3 > T2)
+        ///
+        /// Things to discuss:
+        /// Though the key is expired (not deleted), the commit log do not have delete entry
+        /// So, the latest entry of the key is expired
+        /// And at the moment, we sync the expired key
+        ///
+        /// TODO: ** Needs refactoring:
```

Comment:
@kalluriramkumar In one of the arch calls discuss all of the tickets raised / things we want to refactor for tests.

Reply:
Sure @VJag